### ISSUE TITLE
Gateway compatibility with themes cloning elements with `id` attributes

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#7674]
### Describe the bug

"Vendrify" theme is cloning page elements at checkout.
This poses a problem with our JS code, since we heavily rely on id attribute selectors (or global selectors in general). 
When multiple elements with the same `id` are present on the page, this poses a problem, because the JS might be targeting the wrong element.

A possible solution could be for us to scoping the selectors within a container element.

### To Reproduce

- Insall "Vendrify" theme
- Enable multiple payment methods (e.g.: card, giropay) at checkout
- Attempt to checkout with a valid card

### Actual behavior

Customer is presented with an error message, unable to checkout
![Screenshot 2023-11-09 at 11 51 08 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/f543245d-98a7-415c-8f0b-73ef8273749f)

Warnings are present in the console
![image](https://github.com/Automattic/woocommerce-payments/assets/273592/fcebf2b6-3e38-453b-b605-363f93175fca)

### Expected behavior

Customer is able to checkout

### Additional context

Report from 7267942-zd-a8c

**Comment by pierorocca** [comment]
> "Vendrify" theme is cloning page elements at checkout.

@frosso would you characterize this as a good or bad practice is general? If the latter, I'd be inclined to provide the feedback to the theme developer and put the onus on them to alter their approach.

**Comment by frosso** [comment]
> @frosso would you characterize this as a good or bad practice is general?

It's hard to say if cloning can be considered a "good" or "bad" practice. It isn't something out of the norm for theme developers to duplicate elements on the page to accommodate mobile/desktop design differences.

In WooPay, for example, the "Order summary" section is displayed collapsed at the top of the "account information" section on mobile, but it's displayed expanded "after" the "account information" on desktop. This is done by duplicating the "order summary" section for desktop vs mobile viewports.

I think that in the case of this theme, _maybe_ the cloning of the payment methods section shouldn't be done, because it can contain hidden inputs with sensitive information.
But we (as WooPayments) could also try to minimize the side effects by ensuring that if we use IDs on elements, we instantiate the gateways correctly and target the correct element when needed.

It certainly wouldn't hurt to reach out to the theme developer to ensure the side-effects of their implementation are correctly understood.

**Comment by pierorocca** [comment]
> In WooPay, for example, the "Order summary" section is displayed collapsed at the top of the "account information" section on mobile, but it's displayed expanded "after" the "account information" on desktop. This is done by duplicating the "order summary" section for desktop vs mobile viewports.

Let me rephrase, cloning and reusing the same element ID or not contemplating that element IDs may be in use. If the cloner doesn't need the IDs, they should be removed. I get the convenience of cloning. Not accounting for class or ID collisions isn't ideal.

> But we (as WooPayments) could also try to minimize the side effects by ensuring that if we use IDs on elements, we instantiate the gateways correctly and target the correct element when needed.

Agreed. Can we move away from using IDs in any of the cases above without impact?

Noting a few naming convention inconsistencies. I see a good mix of different naming conventions used for IDs and Classes above - lowercase, BEM, kebab, snake, generic values like "username" and "password".

**Comment by frosso** [comment]
> Let me rephrase, cloning and reusing the same element ID or not contemplating that element IDs may be in use. If the cloner doesn't need the IDs, they should be removed. I get the convenience of cloning. Not accounting for class or ID collisions isn't ideal.

Totally agreed! I was considering that if I were a theme developer, I might not be aware of what other plugins (in this case, WooPayments) might render within certain areas of the screen that I'm cloning, and as a theme developer, I might not be aware that there might have been IDs somewhere within the children of the element I'm cloning.

> Agreed. Can we move away from using IDs in any of the cases above without impact?

We can definitely minimize the number of instances we rely on IDs without impact 👍
